### PR TITLE
Fix litePhysics raycast error

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Summary by CodeRabbit

- **New Features**
	- Enhanced raycasting logic for improved efficiency and accuracy in detecting hits.
  
- **Bug Fixes**
	- Updated test cases to ensure raycasting correctly handles negative distance parameters and edge cases.

### CODE REVIEW

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 1]
```diff
@@ -333,19 +332,17 @@ export class LitePhysicsScene implements IPhysicsScene {
     for (let i = 0, len = colliders.length; i < len; i++) {
       const collider = colliders[i];
 
-      if (collider._raycast(ray, onRaycast, curHit)) {
+      if (collider._raycast(ray, onRaycast, curHit) && curHit.distance < distance) {
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsScene.ts:335]: colliders[i]._raycast

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 2]
```diff
@@ -333,19 +332,17 @@ export class LitePhysicsScene implements IPhysicsScene {
...
-            hitResult.distance = curHit.distance;
-            hitResult.shapeID = curHit.shapeID;
-          } else {
-            return true;
-          }
-          distance = curHit.distance;
+        if (hitResult) {
+          hitResult.normal.copyFrom(curHit.normal);
+          hitResult.point.copyFrom(curHit.point);
+          hitResult.distance = curHit.distance;
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsScene.ts:340]: hitResult.distance = distance =  curHit.distance;

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 3]
```diff
@@ -333,19 +332,17 @@ export class LitePhysicsScene implements IPhysicsScene {
...
+          hitResult.shapeID = curHit.shapeID;
+        } else {
+          return true;
         }
+        distance = curHit.distance;
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsScene.ts:345]: Can delete.

Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsScene.ts:345]:
```
if (colliders[i]._raycast(ray, onRaycast, curHit) && curHit.distance < distance) {
        if (hit) {
          const hitResult = LitePhysicsScene._hitResult;
          hitResult.normal.copyFrom(curHit.normal);
          hitResult.point.copyFrom(curHit.point);
          hitResult.distance = distance = curHit.distance;
          hitResult.shapeID = curHit.shapeID;
          isHit = true;
        } else {
          return true;
        }
      }
```

File: packages/physics-lite/src/LitePhysicsScene.ts [hunk 4]
```diff
@@ -333,19 +332,17 @@ export class LitePhysicsScene implements IPhysicsScene {
     for (let i = 0, len = colliders.length; i < len; i++) {
       const collider = colliders[i];
```
Comment (cptbtptpbcptdtptp) [inline packages/physics-lite/src/LitePhysicsScene.ts:333]: Can delete.

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Review by GuoLei1990 — APPROVED** [review]
